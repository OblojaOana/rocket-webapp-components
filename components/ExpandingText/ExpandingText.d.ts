import { FormControlLabelProps } from '@material-ui/core/FormControlLabel';

export interface ExpandingTextProps extends FormControlLabelProps {
    /**
    * The text content.
    */
    text: string;
    /**
    * @default 2000
    * The minimum length of the narrow text.
    */
    minLength?: number;
    /**
    * @default "Show less"
    * The content of the button when the text is narrow. 
    */
    showLessText: string;
    /**
    * @default "Show more"
    *  The content of the button when the text is wide.
    */
    showMoreText?: string;
}
/**
 *
 * Demos:
 *
 * - https://bit.dev/totalsoft_oss/react-mui/expanding-text
 *
 *  Take a look over the code for further information:https://github.com/osstotalsoft/rocket-webapp-components/blob/master/components/ExpandingText/ExpandingText.js
 */
export default function ExpandingText(props: ExpandingTextProps): JSX.Element;