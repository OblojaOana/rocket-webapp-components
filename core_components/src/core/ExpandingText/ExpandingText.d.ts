export interface ExpandingTextProps {
   /**
   * The text content.
   */
    text: String;
    /**
     * Tooltip reference element.
     */
    minLength?: Number;
    /**
     * Do not respond to focus events.
     */
    showLessText: String;
    /**
     * Do not respond to hover events.
     */
    showMoreText?: String;
}
/**
 *
 * Demos:
 *
 * - [Tooltips](https://material-ui.com/components/tooltips/)
 *
 * API:
 *
 * - [Tooltip API](https://material-ui.com/api/tooltip/)
 */
export default function ExpandingText(props: ExpandingTextProps): JSX.Element;