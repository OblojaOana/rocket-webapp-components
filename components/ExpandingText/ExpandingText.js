import React from 'react';
import PropTypes from 'prop-types';
import { FormControlLabel } from '@material-ui/core';

const ExpandingText = ({ text, minLength, showLessText, showMoreText }) => {

    const [expanded, setExpanded] = React.useState(false);
    const handleChange = _ => {
        setExpanded(!expanded);
    }
    
    return (
        !expanded ?
            <React.Fragment>
                <div>
                    <FormControlLabel
                        control={<div> {text.substring(0, minLength)} <br /> {text.length > minLength ?
                            <div onClick={handleChange} style={{ cursor: "pointer", "color": "blue" }}>{showMoreText}</div> : ""}</div>}
                    />
                </div>
            </React.Fragment> :
            <React.Fragment>
                <div>
                    <FormControlLabel
                        control={<div> {text} <br />
                            <div onClick={handleChange} style={{ cursor: "pointer", "color": "blue" }}>{showLessText}</div></div>}
                    />
                </div>
            </React.Fragment>
    )
}

ExpandingText.defaultProps = {
    minLength: 2000,
    showLessText: "Show less",
    showMoreText: "Show more"
}

ExpandingText.propTypes = {
    text: PropTypes.string.isRequired,
    minLength: PropTypes.number,
    showLessText: PropTypes.string.isRequired,
    showMoreText: PropTypes.string.isRequired
};

export default ExpandingText;