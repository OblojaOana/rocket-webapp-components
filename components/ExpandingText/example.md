##### Copy this snapshot in the overview panel
 ```js
import React from 'react';
import ExpandingText from '@bit/totalsoft_oss.react-mui.expanding-text';

export default (
    <ExpandingText minLength={100}
        text={"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor   incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris   nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."} 
        />
)
 ```