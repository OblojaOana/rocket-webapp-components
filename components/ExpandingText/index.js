import ExpandingText from "../../core/ExpandingText";
export default ExpandingText;
