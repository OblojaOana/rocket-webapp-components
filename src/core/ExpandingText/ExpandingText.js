import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import { FormControlLabel } from '@material-ui/core';

const ExpandingText = ({ text, minLength }) => {

    const [expanded, setExpanded] = React.useState(false);
    const { t } = useTranslation();
    const handleChange = _ => {
        setExpanded(!expanded);
    }

    return (
        !expanded ?
            <React.Fragment>
                <div>
                    <FormControlLabel
                        control={<div> {text.substring(0, minLength)} <br /> {text.length > minLength ?
                            <div onClick={handleChange} style={{ cursor: "pointer", "color": "blue" }}>{t("ShowMore")}</div> : ""}</div>}
                    />
                </div>
            </React.Fragment> :
            <React.Fragment>
                <div>
                    <FormControlLabel
                        control={<div> {text} <br />
                            <div onClick={handleChange} style={{ cursor: "pointer", "color": "blue" }}>{t("ShowLess")}</div></div>}
                    />
                </div>
            </React.Fragment>
    )
}

ExpandingText.defaultProps = {
    minLength: 2000
}

ExpandingText.propTypes = {
    text: PropTypes.string.isRequired,
    minLength: PropTypes.number
};

export default ExpandingText;