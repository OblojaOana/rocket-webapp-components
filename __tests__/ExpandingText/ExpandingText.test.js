import React from 'react';
import renderer from 'react-test-renderer';
import ExpandingText from '../../components/ExpandingText/ExpandingText';

describe('ExpandingText', () => {
    it('snapshot', () => {
        const component = renderer.create(<ExpandingText minLength={100}
            text={"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor   incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris   nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."}
        />);

        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });
});