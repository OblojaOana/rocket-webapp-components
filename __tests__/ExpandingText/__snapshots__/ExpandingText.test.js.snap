// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpandingText snapshot 1`] = `
<div>
  <label
    className="MuiFormControlLabel-root"
  >
    <div>
       
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor   incididunt ut labor
       
      <br />
       
      <div
        onClick={[Function]}
        style={
          Object {
            "color": "blue",
            "cursor": "pointer",
          }
        }
      >
        Show more
      </div>
    </div>
    <span
      className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
    />
  </label>
</div>
`;
